tests/test_posts.py

```diff
@@ -1,58 +1,48 @@
 from fastapi.testclient import TestClient
 from app.main import app
-from app.mongo_db import users_collection
+from tests import utils
 
 client = TestClient(app)
 
-
-# help functions
-def get_auth_header():
-    login_response = client.post(
-        "/login", data={"username": "Test", "password": "1234"}
-    )
-    token = login_response.json().get("access_token")
-    auth_header = {"Authorization": f"Bearer {token}"}
-    return auth_header
-
-
-def get_test_user_id():
-    user = users_collection.find_one({"email": "test@gmail.com"})
-    id = user["_id"]
-    return id
-
+test_user = utils.create_test_user()
+auth_header = utils.get_auth_header(client)
 
 def get_test_post_id():
-    res = client.get("/posts", headers=get_auth_header())
-    return res.json()[0]["id"]
+    res = client.get('/posts',headers=auth_header)
+    post = res.json()[-1]
+    return post["id"]
 
 
 # tests
 def test_create_post():
     res = client.post(
         "/posts",
-        headers=get_auth_header(),
-        json={"title": "Test", "body": "Test", "owner_id": str(get_test_user_id())},
+        headers=auth_header,
```

Comment:
You additionally need to check that such a post was created in the DB and/or make the request for that specific post and see that the application responds with correct data